### PR TITLE
Add memory statistics for the global `ResourceCache`

### PR DESCRIPTION
**Pull request by ptrgags**
Part of https://github.com/CesiumGS/cesium/issues/9886, this PR tracks memory usage in the global `ResourceCache` used to load resources for `ModelExperimental`. It is the global counterpart to #10397 which tracks memory on a per-`ModelExperimental` basis.

Some important caveats:

* This only measures memory loaded through the cache, not all global memory usage in general.
* There is overlap between the memory counted here and the memory counted in #10397. Memory can be shared between multiple tilesets due to the global cache. Meanwhile, a `ModelExperimental` may use a mix of resources loaded from the cache, and additional buffers generated at runtime as needed. Determining the size of the overlap is non-trivial.

[Sandcastle for testing](http://localhost:8080/Apps/Sandcastle/index.html#c=jVRdc+I2FP0rGp5ghgjLkr+6bKYJJFkzMekGsskyvAhbEAVbdiWZAJ3898ofaZM27eTFlq7uOedK50ojpniZwYt9wSTPmNA0vWRUl5IpyARdpSzKE5a+XQdfgZYl+7IUSxHnQmmw4+yZSRMX7BmMGsYfday77MT1fJQbKBdMLju9GtlgINtrJpJuC2qCzQSP5zxlKhSqYLHOZcT3XFTYRpO/xiuhukaj35LGHxLU7HVmXcBgAK4kFboteDIDNDZABXQODnkpAc8FoEoxrZairS/MBUzYmpapPquT5/mWCSO87LDD5HF1FfMbPgnvjiGa8tBI3zrxKHTDbfHwYzQJoEn6PbnamqTwOM0uH6PjIotmiEdP28P1/DuJxpG+ub843ozQdpFdPkX3P/eL7O4wvQ/59WhSLAzZdPxdhVn6mJhxNP/5PJ2foWh+5kyxBRPvIZuuLsPstwt87vDN5GFy5y38MT23vOJcPd2utuib1vNxuV126lNYinUpYl3tVRt/r3OadFO6YmkflDLtgT+WAoDmxHV1mky/t/mdV0x363xQYavBS+UXAINBi4UZ3fOszGaxZEzMChqzCynzqnVIndkaqGImGCxMx3HNd6YVaZJ0W46K8qUx8K+KzfGnbEerbSw7ffC3W7dMGStjBtcyz84qM8Oki3zXCaxeRfSGYaPq3lCfIvCwT2qCd0VsdENxLnmyYZ8iwrb7AZE2x6M+i3d6725UTDMmKVynh3ne2JEwpbmoTwf88s48KrUZUYG7JzZx7MBF0HVtjAghltsHJ8TzAoQQxC4hAfYc0geY2OaPoGcT20UB9t1evxLJJa9eh1akbQOjzSWL/1fZgsQKkIOIhXBAUGBbRuXERBFxfNdFmGAPma9TR31k+8SuFlziBcRvxOuWK/57cxa0fQu7tudi1zHmuajRCCziE0OEPc+2fdfvAwti28J+4GHL9zHGdqvwUn+TUr7u0DJd2PvS6XeGSh9Sdvpaxq88K3KpqxvQhXCgWVak1BQyWJXx1tyAWKnmUgAwHLyFDhO+Azz5+sGLCeLUPERmZV2m6YwfTWudDgcm/1/Q1PQPF5ubHZMpPVRpj+j0uglCCIcDM/0YqfM8XVH5D+Y/AQ) -- In the `Logging` section of the inspector, toggle on the new `ResourceCacheStatistics` like so:
image

@j9liu could you review?

CC @lilleyse

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/resource-cache-statistics/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by j9liu — COMMENTED** [review]
@ptrgags did a first pass and left some comments!

File: Source/Scene/ResourceCache.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import GltfTextureLoader from "./GltfTextureLoader.js";
 import GltfVertexBufferLoader from "./GltfVertexBufferLoader.js";
```
[the remark this answers is not in the document]
Comment (ptrgags) [reply Source/Scene/ResourceCache.js:12]: :thinking: Sort of?

But now I'm realizing I'm forgetting a few cases:

* `GltfStructuralMetadataLoader` loads metadata from `bufferViews`, that's not being accounted for.
* Potentially the buffer could have padding or other data not actually used.

So counting the buffers only would _sometimes_ be more accurate. Though if textures were loaded from a separate image file that's different. And in some cases loaders copy out a slice of data... this is starting to become more tricky.

File: Source/Scene/ResourceCacheStatistics.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+   *
+   * @type {Number}
+   * @private
+   */
+  this.texturesByteLength = 0;
+
+  // Track the sizes of resources by cache key. This is important so
+  // removeLoader() can decrement the count correctly.
+  this._geometrySizes = {};
+  this._textureSizes = {};
```
[the remark this answers is not in the document]
Comment (ptrgags) [reply Source/Scene/ResourceCacheStatistics.js:33]: This time these are dictionaries of `cache key -> size in bytes`, so I'm leaving them as-is.

File: Source/Scene/ResourceCache.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptrgags) [reply Source/Scene/ResourceCache.js:12]: Last week we discussed this offline. It won't be perfect, but the fix we want is to add details for the structural metadata.

**Comment by ptrgags** [comment]
Just leaving another [Sandcastle link](http://localhost:8080/Apps/Sandcastle/index.html#c=zVsJc9u4Ff4rqDZdSxOL5n3Ix1SWldhdH6mt7M42ymQpErIwoUgOSfnYrP57H0CQBKnDjuN2OhnbEvHew8M7PwCMF4Vphu4IvscJOkQhvkcDnJLFXPqVPWuPWx77PojCzCUhTsatXfRt2dkfhzmX5AWR91XyFkmCw2xE5hjkcBn/XATEDU/cDEvTJJqfpZFtykp7HCI0bqmyqnQVpSs7I9nqqVZPsyRZMWVT1y1D023T1h3N+ve4NQ7pbB7TNPVwSCfgc7OvMMj+SgG5nWUSCTMcpiR7BDKQCKOlqvgBhvw21y5/mH/RTkYkwOlZmMbYy6LkgjwQYVpSPKdWuYh8HFQ6eGsFMOmMkimwt4e0E8RI0CWogUDuHU4y7CNqGTQ4OUbRFPV9HO6i3/Ech6hNn8VJdEd8oJo8og8JTt1bknYKrTKcJOATJhVnDatIcULmJCN3OJVc32dGF9xbWzbO2t/oOEKLJOgVJGdReI3TaJF4ufv6KRCe+W3TtGzZ6exSjmVH9M5kQQKfhLfp/14l09KtXCWE5u4DmS/mN16CcXgTux4eJkmU9JCi1pQeh6t+lbJS9eZqGEe+UM+d48QdJW6YTqNkngI105Yz9xDX3cdpRkI3I1HYqy3VBdenkBtaOydESNccy3EcyVZVy9INW90tR3RZN2RVMi0YkE2nHFCAxbJVSbdkx4YRVc1HCjv4JIHFPWPuriypjqYZhg05Z1myocjmrjAISWnCHIZt6ablKEo5JkuWBjyG7oDyiuXIRl2DRfzU1CBc1WXHUnVVMTW6QkG4bmqOotmyrVmGoziGuSuy2ZYCxrI1x9E1WREnXrLfLkniKHmpM3RLcSTFUHRHdjRHdIaqKZaky4qjgwpG5Q3wgCnbEvyCFemW5bzQG7BuRTdNxzEUw5Ft07RFX9i2bdmKDsai3hLCRJbAQqoGSqm2bMpgy+/3ha0rhmxahmoYKvhbdLRumIptqQ48101HFedVbUdX6JpN2VK1VUd4CXSA5IV+sCGsJGgLEAaKY9b8YCg2hIijU41MMSscxVQky1RU2TIt1fiBrLBl2ZE1CEzZtg3HqiUF9ChoYYbiyJasO7aYFKqsqBAfBpiLptULkkJTHMV2DB3cbahmzRE6FAhbU1RYmgX2FnNCMxVZBX1lw4EwNFZd8QMJ4ZiOLdkGZKqmV/rQAXAMTA1msGVNF/MBAliTLAtU0Q3DtF7uB8OERDRsRzMNXYOsE4PeMh3TsHTISkWBpNNqySIblm1pgCggLyBSvt8R4AATfgCJQEmQbdERNtRDamkNijNUIFWcWDF0RbEhLHUVyqamr2TFknWT6SJkJkDT4HGQ95So3Wguu8hfJMxJndxxgCXeBY8oi5CLQuJhFAGSoH2WIohshpEH4EeilAVCYQIlmAOkF77nMnvlp901QdFQRRIGOXmUEIB9nPxbYQHBtysiiqHSXNQNTapFzIeX/C92U2jF77i9SkAwrD2W/vWxf3LdH50Nvpxdfrn6OOLtHmy9pOZudnQpwa7/+AGABEmxBKYL26VH2ryXc6Pzvg9rZUB5JWQYHDnBtwA5Uh4+uiHJuuIUcaGoEuBaKK38uyrLMv1EtSvkz0Af0K+a4MLNZlIWXcNjME3bkCVZpI9J5s02UnfVBjmY95aiZwOm5sgYbQu9tEBDzIKdAsl2u110kz0GmH6qQFFh3XzocBO0Y8NtbtQgqvoCSPEJtX3aQ5+K4Pg0br35dvp+tER/OzxEi9DHU9iF+OjnnxF/foAMuieBjQqV1t75aWpMfdXf6Yxbn3crMfwD3Xk8IfAIDFSTr8gwgSigmMnXXE136Uzl8IunVOT6nMaGOQGYYMt4pTnr61Q3rRPKr+tMN8755Ey5ZEGgqmoTT2u6qJUlC1wjvJ+RDItkn6sC2hHwON8JvX7khZAQS3QIC9u5Ggz7lzv1deia6TsroVbnO+//MmywAaqxPGc726B/PuhfD68+3jSYj4+PzWNlO/P76/5Nkw8asK+b2/neXV0Pb0Z1RlrSwid0PRv9XmdiW/HbxH3czti/+XDaPx9dX/VPVixL/z2Lu8lpar72lHGvLgfXw1HTLxNnoruT7awn17+/v776eNlU2Dm2TOMJhX8bjtbyGq6uaep23pt+k+02CvynU2iNYitJxGv6B+J9pd2HV/UAdsE4dCcBLgYOERVfbvdnbugHzXMjcdd9B+3yNCdq5ycAnhveuamYunOcub6buVcAYgKXHtr4kbeYA6cEkmDjMAzocUjWHrd8cjduiSdPxZmU5MYxDv3BDBpQuyGQ0jceSV7gpumly46qxq2J6331kyget9aQprRmAGhJ41y5cSuMQryFNI5SVkoYrTtJo2CRbaOfRFkWzRm1vIUswNPsKSJwfxzR46+ki6npU3D5s1R2fY46xi09ftimLJjqNomgyA9odDGOnzSZ/tvCxao4iwjGECe4G5CtGk3Bszfkz5xeMbfrFCU+TijmWaS1JeQRnNEAPs3mwQ3dhnqz/fzxnG5JiRuk9cf3mBauxkMihOsMxllxgw7HD32mEKWLBPeq3sfyC5DxLWDlYJC3GrEnsczslLsAfrSJAwDGryiWp6dEj8jCeJH1GbAVEO4cNg40tTjEJVPUruV7p+iPUB/OprDb4CqhezdF4MU7Ei1S2IkINtml2kZsF1I+zWVQ6VxRvpy2wCdx0Z1OtYlYMyx5POzEsZpF9rdwIwGecLpljoELgFwR8gNLMERpJglKzAee3Z39NWzUbMx/hRgCg27oYdiXrYEj73Kq/boKJB3lYOZdoUuxovosAK8Ko8zcFPYwMU6yR6iS6WKSZvTEJc2LpSj6mEP0l8oGOJcLrXza1Fdw4NO1dELvDliybuMoS+Qfb75V8LTc3NKOAiWd7kNPWfqiLlrnM+mxYF7GD388MSevtzDjWlEPXEQhpFllKOsBe3h0kNFdHfxJ4Mc/uuCFp3ewl9GnM/jJjt58Kwx+i7PK4AIWp1CA3oXk3ztLYM+O4FdCf/EZJpH/eCSq1axyVWzW51kbNKhRDTcyczKRc8Uebw/FxXBjUBiexm54uKPuIGb3w50MP2RdFzKXnh2w7fb+ztHB5Oim1PBgb3KUG4+unpd6HsZQHNoEFJX3EYEdTXP94NbwNpvB4Nu3QqCu1Zc6MHdZ7qCmrE/ks5QGxMOCmxDSqi+Vj3y22DULTKjh6PrYDKKcuulhrmpw+XdBLjVBFcvLLfanZt/LQwT+stDcknkkBFx1Oro4p6BvtYWyyRAOUvw96V7ikFrpzatIozQJ3oHuc1oUc9ZZyvI6RSTbSVEYZcgN2FlO2UWlgptKLxjoBrUkmK5MtKlpTMtCXdCJHZdegIa4XetQu5sbVEcQ0+xqNbEX/ffDy1G/6Vv2B34tdzch7tFjjKUL2DgOv1xc/TrsvAgMPNHyCytu6vdruv2K4UvLN0dKe5QD61p8k2ulu/MAe05nj2tt/XWa+pau+6J2uz1NwF83zCDPzpAq+VYteViqKKRHguFBuC+E4VonTBuwhmrm3mGmV4lxORGoVrg2P5KplBJ1WQcUK73WJOP6jK5FkXDQtT7Mylb7vZCyURafrhXr5+/ULb21VpyfXQxLg6/BIrTj0qeI7XwPd/J3FroknEbH0UMXFuAugmxESXaqrrCliRctfHZUhCui+2kKRdjTs5PyIy1H5Zdf3WDBvzHokvejcsb6kcCAKruCUdm2nybthOrEdJa+xFyL/CtbZU1kMU6VTNcDGTbUriPmHC4AfY2/QBHcMRRzsG1ihTsKmPH2LRECdVWVpmTo8wKaoZnzlcQ0cyCgOCHBKUQMgQJEnz8iF4LQDe7dxxThecwvfookqmK8PushDYn+h7OTcQv99ddWqnf9weBpquvfqKwyZcSeSt1FwgUWG9l6i7Dg2IAyxTmrjKzzA2stdKTKZJ9EftHGdQk0Vqvc4mnB1FpjSgjEeQzpnHfc3TUEGTxfLSObsGYBEOnhR3fqzknw2IONTBiltLXv76CMZHT8zbdyAh+nXkJi2rqWOyzBhEF2eFhmHkBoYYz4Sw6jZwziVkN0LcsCswrPmRWWq6Bz+fxaUcF9HZQltfWm5E/cU+T4YefolN5nIsAkLkqie5ZfwjoP9siRUEA2IdrivSNeWocQg3V7bcK1T0Or8+G70ZfB+dngl/L09JT4eYPjFxFlHLK7WQp/72c4ZBRzAFDQmFM2wp7QKowm0cMuvdKlGAsm4nxxKlWvoLFaXb6HJxyUUrVpsvCT0vT4cVAccJbv73Vzg3SbYgBfMHaS4XmbXRY2KaQoZDozfSE5S8RY9ODvwP1L8bj541n9/jCIPHaPTOszu4XhRYRumHog5Ypfc5e3UEDK3NsTlBJKz3MvNqtr5vx3Y9p+/hrRq8zKX0l6xqwD9s7Mq0yav37zjDk/vNYy16/xc/2Vh4Tav9zYtYVDyP+DjcP3n1f+yGnlc4Dlkr/DwCFS5ONNeXIZ5ZX1ua5sOqJsVCuvS6T8frUJeVHjVdSNhFuCL7+7LadEEzeFfQKoOmMnIf+9xdReW3j1BRUNoVxOeYz03BX9uBfE2/ENWXnjhr7npoAv6Ju5oygKJm5ygcNFu6zK1JSbyVhAriO5vQ3w8SLLopC/9z1kVwylYeL8roFbo3b/wJ5UNvFmmG6XS9Nsvq94+VlU876Tz1lkYN7wW7utAybwqJjuH2ROax59V7ktSXvQS2GeDKd7kwWwZ5KXpg38V8LFbFZ99EXcnN9r9ZASP6A0CoiP2D3afpOiC1WGIgXcQ8WnNeeBmSiaX/j1kB0/NGkP9sSlHfjkDhH/cM1/A+B7yXFruggCelk3bgEMA/oV1iBibzVxX1CymXJ0nj+UJOlgD76u58zyCGtI/g8) for testing property tables in the inspector, still have to fix the missing unit tests

**Comment by ptrgags** [comment]
@j9liu updated!

**Review by j9liu — COMMENTED** [review]
@ptrgags -- did another pass. The biggest glaring issue is that the dropdown in the inspector isn't formatted well... maybe something additional needs to be modified in the HTML?

Also this PR should get logged in `CHANGES.md`.

**Comment by ptrgags** [comment]
@j9liu updated!

**Comment by ptrgags** [comment]
@j9liu while trying to fix the broken test, I learned that the metadata buffer views are copied and unloaded from the cache. The overall `StructuralMetadataLoader` is not cached, so really it shouldn't have been counted after all. I'll work on removing it

**Comment by ptrgags** [comment]
@j9liu updated!

**Comment by j9liu** [comment]
@ptrgags - I just fixed a typo in the changelog, otherwise this looks good. I'll wait on Travis to pass before I merge.

**Pull request merged**